src/server/session.ts

```diff
@@ -1895,14 +1900,27 @@ namespace ts.server {
 
         private getFullNavigateToItems(args: protocol.NavtoRequestArgs): readonly NavigateToItem[] {
             const { currentFileOnly, searchValue, maxResultCount } = args;
+            if (!args.file) {

```

Comment:
this is the only substantive change. Everything else is a removal of a frankly ridiculous amount of destructuring/allocation to pass around a project+location pair.